lib/Dancer2/Core/DSL.pm

```diff
@@ -241,11 +242,26 @@ sub _normalize_route {
     my $methods = shift;
     my %args;
 
-    # Options are optional, deduce their presence from arg length.
-    # @_ = ( REGEXP, OPTIONS, CODE )
-    # or
-    # @_ = ( REGEXP, CODE )
-    @args{qw/regexp options code/} = @_ == 3 ? @_ : ( $_[0], {}, $_[1] );
+    # Options are optional, try to deduce their presence from arg length.
+    if ( @_ == 4 ) {
+        # @_ = ( NAME, REGEXP, OPTIONS, CODE )
+        # get 'foo', '/foo', { 'user_agent' => '...' }, sub {...}
+        @args{qw<name regexp options code>} = @_;
+    } elsif ( @_ == 2 ) {
+        # @_ = ( REGEXP, CODE )
+        # get '/foo', sub {...}
+        @args{qw<regexp options code>} = ( $_[0], {}, $_[1] );
```

Comment:
When I first read this, I tripped over `REGEXP, CODE` vs. `regexp options code`.  Suggestion to make this self-documenting and consistent with with the other cases:
```perl
@args{qw<regexp code>} = ( $_[0], $_[1] );
$args{options} = {};
```

Reply:
I prefer not to. Notice that `@args{...}` is always done with an order and I can see the value for each one. Sorry, but this is more readable for me this way.

Reply:
ACK --- understood!